streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -188,17 +190,13 @@ trait ZSink[-R, +E, +A0, -A, +B] { self =>
                 self.initial.flatMap { init =>
                   self
                     .stepChunk(Step.state(init), as.map(ev))
-                    .fold(
-                      e => Step.done((Some(e), f(state._2, b), Step.state(init)), Chunk.empty),
-                      s => Step.leftMap(s)((state._1, f(state._2, b), _))
-                    )
+                    .map(Step.leftMap(_)((f(state._1, b), _)))
```

Comment:
Nice. I noticed that swallowed error too :-)

Reply:
It's now in line with `collectAllWhile`, so that's something. I'm happy.